Warnings in bundled TBB

with
```
#        CMAKE_CXX_COMPILER:     Clang 5.0.1 on platform Linux x86_64
```
I get the following warning:
```
In file included from /ssd/deal-git/bundled/tbb41_20130401oss/src/tbb/itt_notify.cpp:55:
/ssd/deal-git/bundled/tbb41_20130401oss/src/tbb/tools_api/ittnotify_static.c:283:20: warning: passing an object that undergoes default argument promotion to 'va_start' has undefined behavior [-Wvarargs]
    va_start(args, code);
                   ^
/ssd/deal-git/bundled/tbb41_20130401oss/src/tbb/tools_api/ittnotify_static.c:280:49: note: parameter of type '__itt_error_code' is declared here
static void __itt_report_error(__itt_error_code code, ...)
                                                ^
```
Thoughts?

--- a/bundled/tbb41_20130401oss/src/tbb/tools_api/ittnotify_static.c
+++ b/bundled/tbb41_20130401oss/src/tbb/tools_api/ittnotify_static.c
@@ -277,20 +277,24 @@ ITT_EXTERN_C void _N_(error_handler)(__itt_error_code, va_list args);
 #pragma warning(disable: 4055) /* warning C4055: 'type cast' : from data pointer 'void *' to function pointer 'XXX' */
 #endif /* ITT_PLATFORM==ITT_PLATFORM_WIN */
 
-static void __itt_report_error(__itt_error_code code, ...)
-{
+static void __itt_report_error_impl(int code, ...) {
     va_list args;
     va_start(args, code);
     if (__itt_ittapi_global.error_handler != NULL)
     {
-        __itt_error_handler_t* handler = (__itt_error_handler_t*)__itt_ittapi_global.error_handler;
-        handler(code, args);
+        __itt_error_handler_t* handler = (__itt_error_handler_t*)(size_t)__itt_ittapi_global.error_handler;
+        handler((__itt_error_code)code, args);
     }
 #ifdef ITT_NOTIFY_EXT_REPORT
     _N_(error_handler)(code, args);
 #endif /* ITT_NOTIFY_EXT_REPORT */
     va_end(args);
 }
+
+//va_start cannot take enum (__itt_error_code) on clang, so it is necessary to transform it to int
+#define __itt_report_error(code, ...) \
+                __itt_report_error_impl((int)code,__VA_ARGS__)
+
 
 #if ITT_PLATFORM==ITT_PLATFORM_WIN
 #pragma warning(pop)

